sketches/BLE_Scratch/BLE_Scratch.ino

```diff
@@ -113,70 +115,67 @@ uint32_t get_config_bytes(uint8_t *buff, const uint32_t n) {
 }
 
 
-
-void setup() {
-  Serial.begin(9600);
-#ifdef DEBUG
-  while (!Serial)
-    ;
-  Serial.println("Started");
-#endif
+bool init_sensors() {
+  bool ok = true;
 
 #if defined(ARDUINO_NANO_BLE_SENSE)
   if (!APDS.begin()) {
     printSerialMsg("Failed to initialized APDS!");
-
-    while (1)
-      ;
+    ok = false;
   }
 
   if (!HTS.begin()) {
     printSerialMsg("Failed to initialized HTS!");
-
-    while (1)
-      ;
+    ok = false;
   }
 
   if (!BARO.begin()) {
     printSerialMsg("Failed to initialized BARO!");
-
-    while (1)
-      ;
+    ok = false;
   }
 #endif
 
 #if defined(ARDUINO_NANO_BLE_SENSE_R2)
   if (!APDS.begin()) {
     printSerialMsg("Failed to initialized APDS!");
-
-    while (1)
-      ;
+    ok = false;
   }
 
   if (!HS300x.begin()) {
     printSerialMsg("Failed to initialized HTS!");
-
-    while (1)
-      ;
+    ok = false;
   }
 
   if (!BARO.begin()) {
     printSerialMsg("Failed to initialized BARO!");
-
-    while (1)
-      ;
+    ok = false;
   }
 #endif
 
-
-
   // All 3 variants have an IMU on it
   if (!IMU.begin()) {
     printSerialMsg("Failed to initialized IMU!");
+    ok = false;
+  }
+
+  return ok;
+}
+
+void setup() {
+  Serial.begin(9600);
+#ifdef DEBUG
+  while (!Serial)
+    ;
+  Serial.println("Started");
+#endif
 
+  auto ok = init_sensors();
+#if defined(SENSOR_CHECKS)
```

Comment:
I think that since we are checking a single macro, we could use `#ifdef SENSOR_CHECKS`